dlt/destinations/impl/lancedb/lancedb_client.py

```diff
@@ -190,6 +191,29 @@ def upload_batch(
             tbl.merge_insert(
                 id_field_name
             ).when_matched_update_all().when_not_matched_insert_all().execute(records)
+
+            # Remove orphaned parent IDs.
+            if parent_table_name:
+                try:
+                    parent_tbl = db_client.open_table(parent_table_name)
+                    parent_tbl.checkout_latest()
+                except FileNotFoundError as e:
+                    raise DestinationTransientException(
+                        "Couldn't open lancedb database. Batch WILL BE RETRIED"
+                    ) from e
+
+                parent_ids = set(pc.unique(parent_tbl.to_arrow()["_dlt_id"]).to_pylist())
```

Comment:
I wonder if this approach will work well for large dataset..

Reply:
Subqueries don't work so we can't use the datafusion engine here. It will have to be batched